### PR TITLE
Add variant id in ElasticSearch when an equipment is created

### PR DESCRIPTION
**Pull request by niconoir**
Add variant id in ElasticSearch when an equipment is created
Add tombstone or remove equipment from ElasticSearch when equipment is removed in a variant.

### CODE REVIEW

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosService.java [hunk 1]
```diff
@@ -19,9 +21,23 @@
...
+    TombstonedEquipmentInfos addTombstonedEquipmentInfos(@NonNull TombstonedEquipmentInfos tombstonedEquipmentInfos);
 
-    Iterable<EquipmentInfos> findAll(@NonNull UUID networkUuid); // Only for tests
+    void addAllEquipmentInfos(@NonNull final List<EquipmentInfos> equipmentsInfos);
+
+    void addAllTombstonedEquipmentInfos(@NonNull final List<TombstonedEquipmentInfos> tombstonedEquipmentsInfos);
+
+    void deleteEquipmentInfos(@NonNull String equipmentId, @NonNull UUID networkUuid, @NonNull String variantId);
+
+    void deleteVariants(@NonNull UUID networkUuid, List<String> variantIds);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosService.java:34]: Why a list ?

**Comment by SlimaneAmar** [comment]
Remova all StreamSupport.stream ?

**Review by SlimaneAmar — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/dto/TombstonedEquipmentInfos.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode
+@Document(indexName = "#{@environment.getProperty('powsybl-ws.elasticsearch.index.prefix')}tombstoned-equipments")
+@Setting(settingPath = "elasticsearch_settings.json")
+@TypeAlias(value = "TombstonedEquipmentInfos")
+public class TombstonedEquipmentInfos {
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/main/java/org/gridsuite/modification/server/dto/TombstonedEquipmentInfos.java]: Done

File: src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosServiceImpl.java [hunk 3]
```diff
@@ -19,22 +26,87 @@
 public class EquipmentInfosServiceImpl implements EquipmentInfosService {
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosServiceImpl.java:27]: Done

File: src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosService.java (hunk 1 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosService.java:35]: We surely will have to remove multiple variants (when they will be invalidated for example)

File: src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosRepository.java [hunk 4]
```diff
@@ -22,5 +22,11 @@
 public interface EquipmentInfosRepository extends ElasticsearchRepository<EquipmentInfos, String> {
     Iterable<EquipmentInfos> findAllByNetworkUuid(@NonNull UUID networkUuid);
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/main/java/org/gridsuite/modification/server/elasticsearch/EquipmentInfosRepository.java:23]: It is used for unit tests

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**